spec/payout_pal/resource/token_spec.rb

```diff
@@ -0,0 +1,130 @@
+      it "requests the access token from the API" do
+        stub_request(:post, "https://123XYZ:456ABC@api.sandbox.paypal.com/v1/oauth2/token")
+          .with({
+            body: { grant_type: "client_credentials", response_type: "token" },
+            headers: { "Accept" => "application/json", "Content-Type" => "application/x-www-form-urlencoded" }
+          })
+          .to_return(status: 200, body: PayoutPal::Stubs["token"])
+
+        # Ensure HTTP POST request IS being made
+        expect(RestClient).to receive(:post).exactly(1).times.and_call_original
+
+        token = PayoutPal.token
+
+        expect(token.scope).to eq("https://uri.paypal.com/services/subscriptions https://api.paypal.com/v1/payments/.* https://api.paypal.com/v1/vault/credit-card https://uri.paypal.com/services/applications/webhooks openid https://uri.paypal.com/services/invoicing https://uri.paypal.com/payments/payouts https://api.paypal.com/v1/vault/credit-card/.*")
+        expect(token.access_token).to eq("A015z9qL")
+        expect(token.token_type).to eq("Bearer")
+        expect(token.app_id).to eq("APP-80W284485P519543T")
+        expect(token.expires_in).to eq(28800)
+
+        # New token and expiration values are cached
+        expect(PayoutPal.instance_variable_get(:@token)).to eq(token)
+        expect(PayoutPal.instance_variable_get(:@expires_at)).to eq(@local_time + 28800)
+      end
+    end
+
+    context "when an access token exists" do
+      context "when expired" do
+        it "requests a new token" do
+          PayoutPal.instance_variable_set(:@token, @cached_token)
+          PayoutPal.instance_variable_set(:@expires_at, @local_time)
+
+          stub_request(:post, "https://123XYZ:456ABC@api.sandbox.paypal.com/v1/oauth2/token")
+            .with({
+              body: { grant_type: "client_credentials", response_type: "token" },
+              headers: { "Accept" => "application/json", "Content-Type" => "application/x-www-form-urlencoded" }
+            })
+            .to_return(status: 200, body: PayoutPal::Stubs["token"])
+
+          # Ensure HTTP POST request IS being made
+          expect(RestClient).to receive(:post).exactly(1).times.and_call_original
+
+          token = PayoutPal.token
+
+          expect(token.scope).to eq("https://uri.paypal.com/services/subscriptions https://api.paypal.com/v1/payments/.* https://api.paypal.com/v1/vault/credit-card https://uri.paypal.com/services/applications/webhooks openid https://uri.paypal.com/services/invoicing https://uri.paypal.com/payments/payouts https://api.paypal.com/v1/vault/credit-card/.*")
+          expect(token.access_token).to eq("A015z9qL")
+          expect(token.token_type).to eq("Bearer")
+          expect(token.app_id).to eq("APP-80W284485P519543T")
+          expect(token.expires_in).to eq(28800)
+
+          # New token and expiration values are cached
+          expect(PayoutPal.instance_variable_get(:@token)).to eq(token)
```

Comment:
Could we test this stuff through the `PayoutPal.token` interface and ensure cached tokens are returned up until a certain point in time (instead of reaching in and just checking instance variables)?